packages/cli/cli/src/cli.ts

```diff
@@ -204,7 +203,12 @@ function addInitCommand(cli: Argv<GlobalCliOptions>, cliContext: CliContext) {
                 let absoluteOpenApiPath: AbsoluteFilePath | undefined = undefined;
                 if (argv.openapi != null) {
                     if (isURL(argv.openapi)) {
-                        const tmpFilepath = await loadOpenAPIFromUrl({ url: argv.openapi, cliContext });
+                        const result = await loadOpenAPIFromUrl({ url: argv.openapi, logger: cliContext.logger });
+                        if (result.status === LoadOpenApiStatus.Failed) {
+                            cliContext.failAndThrow(result.msg);
+                        }
+
+                        const tmpFilepath = result.msg;
```

Comment:
this is pretty undocumented, its not clear to a reader that `result.msg` is going to have the filepath. the response should have a clear `tempFilepath` property (the discriminated unions will help with this)

Reply:
Thanks for the tip, just pushed a refactor!